tests/test_data_utils.py

```diff
@@ -0,0 +1,324 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import itertools
+import unittest
+
+from datasets import Dataset, DatasetDict
+from parameterized import parameterized
+from transformers import AutoTokenizer
+
+from trl.data_utils import (
+    apply_chat_template,
+    extract_prompt,
+    is_conversational,
+    maybe_apply_chat_template,
+    maybe_extract_prompt,
+    maybe_unpair_preference_dataset,
+    unpair_preference_dataset,
+)
+
+
+class IsConversationalTester(unittest.TestCase):
+    conversational_examples = [
+        {  # Language modeling
+            "messages": [
+                {"role": "user", "content": "What color is the sky?"},
+                {"role": "assistant", "content": "It is blue."},
+            ],
+        },
+        {  # Prompt only
+            "prompt": [{"role": "user", "content": "What color is the sky?"}],
+        },
+        {  # Pompt-completion
+            "prompt": [{"role": "user", "content": "What color is the sky?"}],
+            "completion": [{"role": "assistant", "content": "It is blue."}],
+        },
+        {  # Preference
+            "prompt": [{"role": "user", "content": "What color is the sky?"}],
+            "chosen": [{"role": "assistant", "content": "It is blue."}],
+            "rejected": [{"role": "assistant", "content": "It is green."}],
+        },
+        {  # Preference with implicit prompt
+            "chosen": [
+                {"role": "user", "content": "What color is the sky?"},
+                {"role": "assistant", "content": "It is blue."},
+            ],
+            "rejected": [
+                {"role": "user", "content": "What color is the sky?"},
+                {"role": "assistant", "content": "It is green."},
+            ],
+        },
+        {  # Unpaired preference
+            "prompt": [{"role": "user", "content": "What color is the sky?"}],
+            "completion": [{"role": "assistant", "content": "It is blue."}],
+            "label": True,
+        },
+    ]
+
+    non_conversational_examples = [
+        {"prompt": "The sky is", "completion": " blue."},
+        {"text": "The sky is blue."},
+        {"prompt": "The sky is"},
+        {"prompt": "The sky is", "chosen": " blue.", "rejected": " green."},
+        {"prompt": "The sky is", "completion": " blue.", "label": True},
+    ]
+
+    @parameterized.expand(itertools.product(conversational_examples))
+    def test_conversational(self, example):
+        self.assertTrue(is_conversational(example))
+
+    @parameterized.expand(itertools.product(non_conversational_examples))
+    def test_non_conversational(self, example):
+        self.assertFalse(is_conversational(example))
+
+
+class ApplyChatTemplateTester(unittest.TestCase):
+    tokenizers = [
+        "Qwen/Qwen2-7B-Instruct",
+        # "meta-llama/Meta-Llama-3.1-8B-Instruct", # gated repository. TODO: find a way to test this
```

Comment:
You can use one of the tiny random models from the transformers team, e.g. https://huggingface.co/hf-internal-testing/tiny-random-LlamaForCausalLM

Reply:
Adressed in [040d4cf](https://github.com/huggingface/trl/pull/2020/commits/040d4cf2eac76298898389cca4c7db94b3a8bb1b)